crates/valence_protocol/src/var_int.rs

```diff
@@ -48,16 +48,31 @@ pub enum VarIntDecodeError {
 }
 
 impl Encode for VarInt {
+    // Adapted from VarInt-Simd encode
+    // https://github.com/as-com/varint-simd/blob/0f468783da8e181929b01b9c6e9f741c1fe09825/src/encode/mod.rs#L71
     fn encode(&self, mut w: impl Write) -> anyhow::Result<()> {
-        let mut val = self.0 as u32;
-        loop {
-            if val & 0b11111111111111111111111110000000 == 0 {
-                w.write_u8(val as u8)?;
-                return Ok(());
-            }
-            w.write_u8(val as u8 & 0b01111111 | 0b10000000)?;
-            val >>= 7;
-        }
+        let x = unsafe { std::mem::transmute::<i32, u32>(self.0) } as u64;
```

Comment:
If I'm not mistaken, this should be equivalent to `self.0 as u64`.

Reply:
Yea that's correct. I should have noticed that LOL.